app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -186,18 +189,19 @@ def download_statistics
 
   def unsubmit
     authorize!(:update, @assessment)
-    submission = @assessment.submissions.find(params[:submission_id])
-    success = submission.transaction do
-      submission.update!('unmark' => 'true') if submission.graded?
-      submission.update!('unsubmit' => 'true')
+    @submission = @assessment.submissions.find(params[:submission_id])
+    success = @submission.transaction do
+      @submission.update!('unmark' => 'true') if @submission.graded?
+      @submission.update!('unsubmit' => 'true')
+      monitoring_service&.continue_listening!
```

Comment:
Would this and line 226 below be run? seems like `should_monitor?` would return false since its instructors who unsubmit or delete student's submissions.

Reply:
Perhaps instead of continuing to listen, why not update the status to listening when users access the edit page and the submission is in attempting state?

Reply:
>Would this and line 226 below be run?

You're right, it won't. I'll make the correct changes.

> Perhaps instead of continuing to listen, why not update the status to listening when users access the edit page and the submission is in attempting state?

Because we are able to stop sessions remotely. If every attempting load changes the associated session to `:listening`, then (a) we won't have this remote control, (b) at some point, every session will become `:listening`.